crates/core/src/buc/store/util/path.rs

```diff
@@ -0,0 +1,139 @@
+use core::fmt;
+use std::{
+	borrow::Borrow,
+	hash::{Hash, Hasher},
+	ops::Deref,
+};
+
+use crate::sql::Value;
+
+/// Path represents a normalized file path in the object store.
+#[derive(Clone, Debug)]
+pub struct Key(String);
```

Comment:
Shall we use a less generic name? PathKey rather than Key?

Reply:
Renamed to `ObjectKey` in [120e55e](https://github.com/surrealdb/surrealdb/pull/5701/commits/120e55e0ad47bb91671a7d96931d22c5488e2b96)